docs/data/routes/docs/nextjs/creating-components/layout-concepts/en.md

```diff
@@ -0,0 +1,125 @@
+title: Layout & Component Concepts
+---
+# Key Concepts: Layout, Components, Component Factory
+
+## Layout
+### Sitecore "Layout" vs Next.js "Layout"
+Recall that Content Authors assemble page layout by adding renderings to placeholders (see [content authoring concepts](/docs/nextjs/introduction/content-authoring-concepts)). To be editable, a page must define at least one root (i.e. page-level) placeholder. So, in the context of Sitecore, "Layout" refers to a page's placeholder and renderings hierarchy. Layout data in defined by Content Authors through a WYSIWYG UI, and under the hood it's saved to the database as XML data. JSS apps never work with this XML data; that is purely for powering Sitecore's authoring interface. When a JSS app needs layout data, it uses a Sitecore endpoint to fetch the data in JSON format.
+
+In the context on Next.js, "Layout" is a naming convention for a [top-level component that's shared by all pages](https://nextjs.org/learn/basics/assets-metadata-css/layout-component).
+
+JSS merges these concepts, utilizing a [shared React component named "Layout"](https://github.com/Sitecore/jss/blob/4db9ab5831d9bfb31c20051deb1dfb02b2027968/samples/nextjs/src/Layout.tsx#L94) to define root placeholders. This serves as the entry point for rendering the components hierarchy based on how Content Authors assembled renderings in the WYSIWYG editor.
+
+At first, it may seem redundant to have to define components twice (as Sitecore renderings and as React components). But remember that there are multiple options for implementing web apps on Sitecore, and JavaScript is just one of the choices. Sitecore saves layout data as XML because it's implementation agnostic. Multiple apps using different implementation approaches could coexist in the same Sitecore instance, and Content Authors would have the same authoring experience for all of them. This architecture makes managing content easier for business users.
+
+### Adding more root placeholders
+You are free to define multiple root placeholders. To do so, add additional <Placeholder> instances to `Layout`, giving each placeholder a *unique name*. Then add all root placeholder names to the `config.rootPlaceholders` array in `package.json`. On initial page load, Sitecore uses this value to determine what layout data to pass to the Next.js app
+
+> Note: A common pattern is to define root placeholders for Header, Main Content, and Footer. Front-end developers and Sitecore developers should work together to align on what is best, based on business needs.
+
+### Varying root placeholders by page template
+Sometimes, it may be necessary to render different root placeholders depending on the page type. Since React components can return different elements conditionally, this can be achieved easily by returning different <Placeholder> components based on the value of `props.layoutData.sitecore.route.templateName` or `props.layoutData.sitecore.route.templateId`.
```

Comment:
Is this a common scenario?

Reply:
It's common in MVC implementations (you would have different Layout items pointing to different Views), and I've been asked how to achieve the same in JSS. So it's at least come up; not sure exactly how common it is.

Reply:
@nickwesselman  - Ok to keep this content?